packages/grant-explorer/src/features/round/ViewCartPage/SummaryContainer.tsx

```diff
@@ -0,0 +1,392 @@
+import { ChainId, PassportState } from "common";
+import { useCartStorage } from "../../../store";
+import React, { useEffect, useMemo, useState } from "react";
+import { BigNumber, ethers } from "ethers";
+import { Summary } from "./Summary";
+import ErrorModal from "../../common/ErrorModal";
+import ChainConfirmationModal from "../../common/ConfirmationModal";
+import { ChainConfirmationModalBody } from "./ChainConfirmationModalBody";
+import { CartProject, ProgressStatus } from "../../api/types";
+import { useQFDonation } from "../../../context/QFDonationContext";
+import { modalDelayMs } from "../../../constants";
+import { useNavigate } from "react-router-dom";
+import { useAccount, useNetwork, useSigner, useSwitchNetwork } from "wagmi";
+import { Logger } from "ethers/lib.esm/utils";
+import { datadogLogs } from "@datadog/browser-logs";
+import { Button } from "common/src/styles";
+import { InformationCircleIcon } from "@heroicons/react/24/solid";
+import { usePassport } from "../../api/passport";
+import useSWR from "swr";
+import _, { round } from "lodash";
+import { getRoundById } from "../../api/round";
+import MRCProgressModal from "../../common/MRCProgressModal";
+import { MRCProgressModalBody } from "./MRCProgressModalBody";
+import { JsonRpcSigner } from "@ethersproject/providers";
+
+export function SummaryContainer() {
+  const { chain } = useNetwork();
+  const { projects } = useCartStorage();
+  const payoutTokens = useCartStorage((state) => state.chainToPayoutToken);
+  const projectsByChain = _.groupBy(projects, "chainId") as {
+    [chain: number]: CartProject[];
+  };
+
+  const { data: rounds } = useSWR(projects, (projects) => {
+    const uniqueProjects = _.uniqBy(projects, "roundId");
+    return Promise.all(
+      uniqueProjects.map((proj) => getRoundById(proj.roundId, proj.chainId))
+    );
+  });
+
+  /** The ids of the chains that will be checked out */
+  const [chainIdsBeingCheckedOut, setChainIdsBeingCheckedOut] = useState<
+    ChainId[]
+  >(Object.keys(projectsByChain).map(Number));
+
+  const [checkedOutChainIds, setCheckedOutChainIds] = useState<ChainId[]>([]);
+
+  /** The ID of the current chain (from wallet) */
+  const currentChainId = (chain?.id ?? chainIdsBeingCheckedOut[0]) as ChainId;
+  const { data: signer } = useSigner();
+
+  const { switchNetworkAsync } = useSwitchNetwork({
+    onSuccess: () => {
+      setChainSwitched(true);
+    },
+  });
+
+  /** The current payout token  */
+  const currentPayoutToken = payoutTokens[currentChainId];
+
+  /** We find the round that ends last, and take its end date as the permit deadline */
+  const currentPermitDeadline =
+    rounds
+      ?.sort((a, b) => a.roundEndTime.getTime() - b.roundEndTime.getTime())[0]
```

Comment:
`sort` modifies the array it's called on, please double-check that this doesn't mess with SWR internal cache or smething.

Reply:
Good catch, god bless mutability. 
No idea if it does, don't have time to find out, preemptively fixed by "cloning"